### PR TITLE
HasDerivation should be called MayHaveDerivation #3581

### PR DESCRIPTION
**Pull request by daijingz**
My Pull request for issue #3581

I feel there are not too many things that need to be mentioned here. What kind of information should I include here?

Assume all mentioned "HasDerivation" places are relevant to the target class "HasDerivation" so that those instances are required to be modified

### CODE REVIEW

File: code/drasil-theory/lib/Theory/Drasil/DataDefinition.hs [hunk 1]
```diff
@@ -87,7 +87,7 @@ instance HasDecRef          DataDefinition where getDecRefs = ddPkt pktDR
 -- | Equal if 'UID's are equal.
 instance Eq                 DataDefinition where a == b = (a ^. uid) == (b ^. uid)
 -- | Finds the derivation of the 'DataDefinition where'. May contain Nothing.
-instance HasDerivation      DataDefinition where derivations = ddPkt pktMD
+instance MayHaveDerivation      DataDefinition where derivations = ddPkt pktMD
```
Comment (balacij) [inline code/drasil-theory/lib/Theory/Drasil/DataDefinition.hs:90]: Minor code style issue: we align the type arguments of lists of instances.

**Review by balacij — CHANGES_REQUESTED** [review]

File: code/drasil-theory/lib/Theory/Drasil/DataDefinition.hs (hunk 1 above)
Comment (daijingz) [reply code/drasil-theory/lib/Theory/Drasil/DataDefinition.hs:90]: Do not understand what should I change, do you mean this? @balacij (I use the above example)
instance HasDerivation               DataDefinition where derivations = ddPkt pktMD
instance MayHaveDerivation      DataDefinition where derivations = ddPkt pktMD <-- now there argument types are aligned

Comment (balacij) [reply code/drasil-theory/lib/Theory/Drasil/DataDefinition.hs:90]: Just so you know, you can wrap code with 3 backticks to make a code block. Using the code block will use a monospaced font, so seeing the spacing is easier :smile: 

I _think_ that looks right. The point is that the `DataDefinition`s should be vertically aligned there.

Comment (daijingz) [reply code/drasil-theory/lib/Theory/Drasil/DataDefinition.hs:90]: So what should I do is just aligning all ```DataDefinition```
Here is an example (as stated beyond):
```
instance HasDerivation               DataDefinition where derivations = ddPkt pktMD
instance MayHaveDerivation           DataDefinition where derivations = ddPkt pktMD
```
Now it is aligned

Comment (balacij) [reply code/drasil-theory/lib/Theory/Drasil/DataDefinition.hs:90]: Yes, but that looks like you added too much space. Align the `DataDefinition where...` onwards with the others listed in the files, please.

Comment (daijingz) [reply code/drasil-theory/lib/Theory/Drasil/DataDefinition.hs]: Sorry for the delay. (Working on other things) In this modification, I changed three files: 
```code/drasil-theory/lib/Theory/Drasil/DataDefinition.hs```
```code/drasil-theory/lib/Theory/Drasil/GenDefn.hs```
```code/drasil-theory/lib/Theory/Drasil/InstanceModel.hs```

They have similar alignment problems, but now all these issues have been solved. As you can see: https://github.com/JacquesCarette/Drasil/pull/3653/files#diff-bab435096d39e524c93900a45bea3ad6e1b439001c0a54ea1f27d94d28bd824e

**Review by balacij — APPROVED** [review]
Thanks, @daijingz!

**Pull request closed without merging**